src/tests/dmstest/main.cpp

```diff
@@ -32,11 +33,17 @@ using namespace bson ;
 using namespace std ;
 
 UINT32  TESTTHREADS = 10 ;
-UINT32  LOOPNUM     = 100000 ;
+UINT32  LOOPNUM     = 10000 ;
 UINT32  TESTCREATETABLES = 10 ;
 CHAR *  defaultCollectionName = "foo" ;
 CHAR *  COLLECTIONNAME = defaultCollectionName ;
 dmsBlockUnit *myUnit = NULL ;
+dmsMBContext *context = NULL ;
+dmsRecordGenerator generator ;
+dmsExtScannerBase *scanner ;
+dmsRecordID recordID ;
+ossValuePtr recordDataPtr = 0 ;
+
 BSONObj sampleObj = BSON("name"<<"insert block data") ;
 
 int main ()
```

Comment:
can you package main into 3 separate parts, each of them has its function/interface
1. prepare: do create necessary files
2. insert
3. query
Then you can do run the program with different parameters to trigger different functionalities.